### PR TITLE
Updates to DumpUtility

### PR DESCRIPTION
**Pull request by cljohnso**
This commit stream updates `DumpUtility` to:

* correct recognition `java.nio.ByteBufferAsIntBufferB` (typo)
* make adjustments to support Java 17+
* add usage comments
* add tests
* minor updates to output for `LongBuffer` and `IntBuffer`

The commit stream also adds the `ModuleSupport` class providing support for checking if, under Java 9+ the UNNAMED module has reflection access to a given class/instance.

### CODE REVIEW

**Comment by cljohnso** [comment]
Seems I have a toolchain issue with CI machines ... will need to coordinate with @akomakom

**Comment by cljohnso** [comment]
> Seems I have a toolchain issue with CI machines ... will need to coordinate with @akomakom

Ok ... Azure CI appears to lack support for `toolchains.xml` -- which the current approach to testing `DumpUtility` needs.  I need to figure out a different approach.

**Comment by cljohnso** [comment]
> > Seems I have a toolchain issue with CI machines ... will need to coordinate with @akomakom
> 
> Ok ... Azure CI appears to lack support for `toolchains.xml` -- which the current approach to testing `DumpUtility` needs. I need to figure out a different approach.

I changed the approach to avoid dependence on `toolchains.xml`.  The "extra" test execution for `DumpUtilityTest` now runs under the JVM running the Maven task.  If the current JVM is less than 17, the extra test execution is suppressed; if 17 or greater, `DumpUtilityTest` is run with the proper `--add-opens` set.

File: test-tools/src/main/java/org/terracotta/utilities/test/ModuleSupport.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    return isOpen(ref.getClass());
+  }
+
+  /**
+   * Determines if the module and package of the class provided is <i>open</i> to the <i>unnamed</i> module
+   * for this class.
+   *
+   * @param targetClass class for which "openness" is checked
+   * @return {@code true} if {@code targetClass} is open to this unnamed module; {@code false} otherwise;
+   * is an error is raised while checking the status, the error is recorded to {@code System.err}
```
Comment (prasantakumar82) [inline test-tools/src/main/java/org/terracotta/utilities/test/ModuleSupport.java:90]: Same as previous

File: test-tools/src/main/java/org/terracotta/utilities/test/ModuleSupport.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    }
+  }
+
+  /**
+   * Determines if the module and package of the object provided is <i>open</i> to the <i>unnamed</i> module
+   * for this class.
+   *
+   * @param ref object for which "openness" is checked
+   * @return {@code true} if {@code ref} is open to this unnamed module; {@code false} otherwise;
+   * is an error is raised while checking the status, the error is recorded to {@code System.err}
```
[the remark this answers is not in the document]
Comment (cljohnso) [reply test-tools/src/main/java/org/terracotta/utilities/test/ModuleSupport.java]: 'is' -> 'if'

**Review by prasantakumar82 — APPROVED** [review]

**Review by devlg75 — APPROVED** [review]

**Pull request merged**